### PR TITLE
refactor: deduplicate utility functions into lib.findings (#232)

### LINKED ISSUE #232: Deduplicate shared utility functions across council scripts
## Context

Surfaced during [PR #231](https://github.com/misty-step/cerberus/pull/231) hindsight review. The finding-aggregation consolidation addressed the grouping loop duplication but left several utility functions duplicated across files.

## Duplicated functions

| Function | Locations | Notes |
|----------|-----------|-------|
| `as_int` / `_as_int` | `findings.py`, `render_council_comment.py`, `post-council-review.py` | Identical logic |
| `normalize_severity` / `_normalize_sev` | `findings.py`, `render_council_comment.py`, `post-council-review.py` | Identical logic (findings.py variant collapses whitespace) |
| `SEVERITY_ORDER` / `_SEVERITY_ORDER` | `findings.py`, `render_council_comment.py`, `post-council-review.py` | Same dict literal in three files |
| `split_reviewer_description` | `render_council_comment.py`, `post-council-review.py` | Identical |
| `reviewer_label` | `render_council_comment.py`, `post-council-review.py` | Nearly identical |

## Proposed approach

Move canonical versions to `lib/findings.py` (or a new `lib/council_helpers.py`) and have callers import them. This is a pure refactor with no behavior change.

## Non-goal

Don't over-abstract. These are small utility functions — the goal is one canonical copy, not a framework.

### PR DESCRIPTION
**Pull request by moomooskycow**
## Summary

Closes #232

`as_int`, `normalize_severity`, `SEVERITY_ORDER`, `split_reviewer_description`, and `reviewer_label` were duplicated verbatim across `findings.py`, `render_council_comment.py`, and `post-council-review.py`. Each copy had minor naming drift (`_as_int` vs `as_int`, `_SEVERITY_ORDER` vs `SEVERITY_ORDER`) with no behavioral difference — just maintenance risk and cognitive overhead when tracing which copy was authoritative.

This PR promotes canonical versions to `lib/findings.py` as the single source of truth and deletes the duplicates from the two callers. Pure refactor: no behavior change.

## Changes

- **`scripts/lib/findings.py`** — added public aliases (`as_int`, `SEVERITY_ORDER`) and moved canonical implementations of `normalize_severity`, `split_reviewer_description`, and `reviewer_label` here
- **`scripts/lib/render_council_comment.py`** — removed 4 local duplicates; updated imports
- **`scripts/post-council-review.py`** — removed 4 local duplicates; updated imports; swapped `_SEVERITY_ORDER` refs to the imported `SEVERITY_ORDER`
- **`tests/test_findings.py`** — 89 new lines covering the newly-public API: `as_int`, `normalize_severity`, `SEVERITY_ORDER`, `split_reviewer_description`, `reviewer_label`

## Acceptance Criteria

From issue #232:
- [x] `as_int` has one canonical copy in `lib/findings.py`; callers import it
- [x] `normalize_severity` has one canonical copy in `lib/findings.py`; callers import it
- [x] `SEVERITY_ORDER` dict literal exists once; callers import it
- [x] `split_reviewer_description` has one canonical copy; callers import it
- [x] `reviewer_label` has one canonical copy; callers import it
- [x] No behavior change — pure refactor
- [x] Not over-abstracted — no new framework, just moved utilities

## Manual QA

```bash
cd scripts
pip install pytest pytest-cov pyyaml

# All tests pass
python3 -m pytest ../tests/ -v

# Coverage still meets the 70% gate
python3 -m pytest ../tests/ --cov=. --cov-report=term-missing

# Verify no stale local definitions remain
grep -n "def as_int\|def normalize_severity\|_SEVERITY_ORDER\s*=" \
  lib/render_council_comment.py post-council-review.py
# Expected: no output (all removed)

# Verify imports are wired
grep -n "from lib.findings import" lib/render_council_comment.py post-council-review.py
# Expected: both files import the consolidated symbols
```

## Before / After

**Before:** `as_int`, `normalize_severity`, `SEVERITY_ORDER`, `split_reviewer_description`, and `reviewer_label` each existed in 2-3 files with slight naming variations. Internally, `findings.py` used alias patterns (`_as_int` + `as_int = _as_int`, `_SEVERITY_ORDER` + `SEVERITY_ORDER = _SEVERITY_ORDER`) and had two severity normalizers with identical logic (`_normalize_sev` and `normalize_severity`). Tests used per-method imports.

**After (post-polish):** Single canonical definitions with no alias indirection. `normalize_severity` accepts an optional `order` parameter, eliminating the internal `_normalize_sev` duplicate. Tests use module-level imports. 7 new edge-case tests cover boolean coercion, numeric severity input, custom order parameter, multiple separators, and mixed-case reviewer names.

No user-facing output change — this is internal plumbing.

## Test Coverage

New tests in `tests/test_findings.py` (66 total, from 22 pre-PR):
- `TestAsInt` — None, int, string-int, invalid string, float truncation, boolean coercion
- `TestNormalizeSeverity` — known severities, unknown->info fallback, whitespace collapse, numeric input, custom order
- `TestSeverityOrder` — public constant shape
- `TestSplitReviewerDescription` — em-dash, hyphen, no separator, empty/None, multiple em-dashes, multiple hyphens
- `TestReviewerLabel` — description role wins, perspective fallback, codename fallback, unknown perspective skipped, empty dict, mixed-case name, underscore perspective

Every newly-public symbol has dedicated test coverage including edge cases. `findings.py` is at 100% line+branch coverage.

## Summary by CodeRabbit

* **Refactor**
  * Consolidated shared utilities into a centralized module to ensure consistent severity normalization, ordering, and reviewer label/display across the app.
  * Improved reviewer description parsing for clearer role/tagline presentation and more robust name capitalization.
  * Exposed a public numeric-coercion helper for consistent handling of numeric fields.

* **Tests**
  * Added comprehensive tests covering severity handling, reviewer formatting/parsing, ordering, and numeric coercion to improve reliability.

### CODE REVIEW

File: scripts/lib/findings.py [hunk 1]
```diff
@@ -53,11 +55,51 @@ def _as_int(value: object) -> int | None:
...
+as_int = _as_int
+
+
 def _normalize_sev(value: object, order: dict[str, int]) -> str:
     text = " ".join(str(value or "").strip().lower().split())
     return text if text in order else "info"
 
 
+def normalize_severity(value: object) -> str:
+    """Normalize severity string; unknown values collapse to 'info'."""
```
[the remark this answers is not in the document]
Comment (phrazzld) [reply scripts/lib/findings.py]: Fixed in 0c0b317. Reverted both `normalize_severity` and `_normalize_sev` to plain `.strip().lower()`, matching the original implementations exactly. All 1129 tests pass.

**Comment by phrazzld** [comment]
## PR Unblocked

**Conflicts**: none
**CI**: green (12/12 checks passing including Council Verdict)
**Reviews**: 1 addressed

The open thread (`normalize_severity` whitespace behavior) was resolved in 0c0b317: reverted both `normalize_severity` and `_normalize_sev` to plain `.strip().lower()`, matching the original implementations exactly and making the pure-refactor claim accurate. Thread replied and resolved.

Ready for re-review.
